commands/core/queue.drush.inc

```diff
@@ -116,8 +116,11 @@ function drush_queue_run($queue_name) {
   $end = time() + (isset($info['cron']['time']) ? $info['cron']['time'] : 15);
   $queue = drush_queue_get($queue_name);
   while (time() < $end && ($item = $queue->claimItem())) {
-    $function($item->data);
+    $result = $function($item->data);
     $queue->deleteItem($item);
+    if (isset($result) && $result === FALSE) {
+      drush_set_error('DRUSH_QUEUE_RUN_ERROR', dt('Error during queue run.'));
```

Comment:
This is going to give the same error for each item that fails. Do we want one error? if not we should ideally state which item is failing. Also, if we rely on the exception technique in D8 we will only get one error message.